mono/utils/mono-mmap.c

```diff
@@ -305,7 +305,7 @@ mono_valloc (void *addr, size_t length, int flags, MonoMemAccountType type)
 		}
 		if ((flags & MONO_MMAP_JIT) && (use_mmap_jit || is_hardened_runtime == 1))
 			mflags |= MAP_JIT;
-#if defined(HOST_ARM64)
+#if defined(TARGET_OSX) && defined(HOST_ARM64)
 		/* Patching code on apple silicon seems to cause random crashes without this flag */
```

Comment:
Why is this needed, isn't the __builtin_available (macOS 11, *) check enough ?

Reply:
No it's not enough, because the `*` means any non-mentionned platform will have it enabled. I had a crash on boot on iOS and tvOS because this section set the `MAP_JIT` on those platforms and it's not allowed.

Reply:
isn't this whole if supposed to only be executed on macOS? it checks `if (get_darwin_version () >= DARWIN_VERSION_MOJAVE)`, maybe we should add the TARGET_OSX there?

Reply:
Sure that would make sense as well; I tried to keep my changes to a minimum, but in that case it's indeed more logical to restrict the whole segment. I'll try to do that change today

Reply:
Fixes at 5cf2adf3c75ae7d9c58174256af09ac1a5334f2b